Userland/Libraries/LibPDF/Parser.cpp

```diff
@@ -422,9 +423,27 @@ NonnullRefPtr<StringObject> Parser::parse_string()
 {
     ScopeGuard guard([&] { consume_whitespace(); });
 
-    if (m_reader.matches('('))
-        return make_object<StringObject>(parse_literal_string(), false);
-    return make_object<StringObject>(parse_hex_string(), true);
+    String string;
+    bool is_binary_string;
+
+    if (m_reader.matches('(')) {
+        string = parse_literal_string();
+        is_binary_string = false;
+    } else {
+        string = parse_hex_string();
+        is_binary_string = true;
+    }
+
+    if (static_cast<u8>(string[0]) == 0xfe && static_cast<u8>(string[1]) == 0xff) {
+        // The string is encoded in UTF16-BE
+        string = TextCodec::decoder_for("utf-16be")->to_utf8(string.substring(2));
+    } else if (static_cast<u8>(string[0]) == 239 && static_cast<u8>(string[1]) == 187 && static_cast<u8>(string[2]) == 191) {
```

Comment:
Possibly less annoying to read:
```suggestion
    } else if (string.bytes().starts_with(Array<u8, 3> { 239, 187, 191 }) {
```

I'm not too sure if `string.bytes()` returns a `Span<u8>` though, so this might not be correct.

Reply:
Oh that is much nicer, didn't know that was a thing (it was correct).